MoltenVK/MoltenVK/GPUObjects/MVKDevice.mm

```diff
@@ -1607,6 +1626,16 @@ static uint32_t mvkGetEntryProperty(io_registry_entry_t entry, CFStringRef prope
 }
 #endif	//MVK_IOS
 
+#if MVK_TVOS
+void MVKPhysicalDevice::initGPUInfoProperties() {
+  uint32_t devID = 0;
```

Comment:
No idea how to generate device IDs here. I have access to a few different models of Apple TVs if some wants to point me in the right direction.

Reply:
For iOS, MoltenVK uses a scheme where the device ID is `0xaXXY`, where "XX" is Apple's chip number as a BCD integer (e.g. `0xa080` for A8, `0xa120` for A12) and "Y" is incremented to indicate variations--A8 and A9, for example, have `0xa080` and `0xa090` respectively for single-core iPhones and `0xa081` and `0xa091` for multicore iPhones.

Reply:
Yeah. Given tvOS's use of the same SoC's as iOS, we can follow a similar pattern, based on tvOS feature sets.

Reply:
Okay, so according to https://developer.apple.com/metal/Metal-Feature-Set-Tables.pdf:

`tvOS GPU family 1` = A8
`tvOS GPU family 2` = A9, A10

How can I figure out where the switch from A9 to A10 happens on the family2?

Reply:
The Apple TV HD has an A8 (src: https://www.apple.com/apple-tv-hd/specs/)

The Apple TV 4K has an A10X (src: https://www.apple.com/apple-tv-4k/specs/)

Reply:
As you can see, on iOS I finagled the `X` part from core count. That part at least should translate over to tvOS.

Hmmm...the A9/A10 split seems to be arbitrary. See [this document](https://developer.apple.com/documentation/metal/mtlfeatureset?language=objc). 

I'm not sure if it's trying to indicate that either or both SoC could be used on either or both platforms, or  the A10 should only be on tvOS and the A9 is iOS. In which case, the iOS setting might be in error here. We'll have to look at some iOS device specs I guess.

Reply:
Since the only two models are A8 and A10X, it didn't seem necessary to differentiate further. If you'd like I can add the X bit, but seems like that could just be set always and the core count check is not applicable in this case.

Reply:
Okay. Whatever works. If it's either A8 or A10X, you can set both the number and the X-bit from the GPU feature set alone.

Reply:
Great, done!